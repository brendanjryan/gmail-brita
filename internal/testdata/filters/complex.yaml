emails:
  - me@example.com
  - other@example.com

filters:
  - name: Complex Test Filter
    conditions:
      has:
        - test:condition
        - from:test@example.com
      has_not:
        - to:me@example.com
        - cc:me@example.com
    actions:
      label: test-label
      archive: true
      mark_read: true
      star: true
      never_spam: true

  - name: List Filter
    conditions:
      has:
        - list:test@example.com
    actions:
      label: test-list
      archive_unless_directed:
        mark_read: true 
