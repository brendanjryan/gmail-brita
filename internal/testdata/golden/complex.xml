<?xml version="1.0" encoding="UTF-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xmlns:apps="http://schemas.google.com/apps/2006">
  <title>Mail Filters</title>
  <entry>
    <category term="filter"></category>
    <title>Mail Filter</title>
    <apps:property name="hasTheWord" value="test:condition from:test@example.com"/>
    <apps:property name="doesNotHaveTheWord" value="to:me@example.com cc:me@example.com"/>
    <apps:property name="label" value="test-label"/>
    <apps:property name="shouldArchive" value="true"/>
    <apps:property name="shouldMarkAsRead" value="true"/>
    <apps:property name="shouldStar" value="true"/>
    <apps:property name="neverSpam" value="true"/>
  </entry>
  <entry>
    <category term="filter"></category>
    <title>Mail Filter</title>
    <apps:property name="hasTheWord" value="list:test@example.com"/>
    <apps:property name="label" value="test-list"/>
    <apps:property name="shouldArchive" value="true"/>
    <apps:property name="shouldMarkAsRead" value="true"/>
  </entry>
</feed> 
